### PR TITLE
fix #34

### LINKED ISSUE #34: Provide option to supply output and input file name in generate_ics.py
Implement optional CLI arguments `--output` and `--input` using which the user can supply the path to the output ICS file and the input data file

### PR DESCRIPTION
**Pull request by rohitner**
now we can provide input and output through cli

### CODE REVIEW

**Comment by rohitner** [comment]
ping! @nishnik

File: generate_ics.py [hunk 1]
```diff
@@ -73,10 +81,10 @@ def main():
     now = datetime.datetime.now()
 
     # Get your timetable
-    with open('data.txt') as data_file:    
+    with open(args.input) as data_file:
```
[the remark this answers is not in the document]
Comment (nishnik) [reply generate_ics.py]: This would also be the right PR to do error handling and check if `data.txt` exists or not.

**Comment by rohitner** [comment]
I have done error handling for the two possible cases.

File: generate_ics.py [hunk 2]
```diff
@@ -73,10 +81,17 @@ def main():
     now = datetime.datetime.now()
 
     # Get your timetable
-    with open('data.txt') as data_file:    
+    if not os.path.exists("data.txt"):
```
Comment (icyflame) [inline generate_ics.py]: Hey, this is a good check and it should be there, but it shouldn't be here. This should all happen alongwith the argument parsing.

The best things to do is parse the arguments, and set two variables `INPUT_FILE` and `OUTPUT_FILE`. Then, before starting the script, check if `INPUT_FILE` exists before starting execution.

Comment (rohitner) [reply generate_ics.py]: Done :+1:

File: generate_ics.py [hunk 3]
```diff
@@ -10,12 +10,31 @@
...
+parser.add_argument("-o", "--output")
+args = parser.parse_args()
+
 DEBUG = False
 GENERATE_ICS = True
 TIMETABLE_DICT_RE ='([0-9]{1,2}):([0-9]{1,2}):([AP])M-([0-9]{1,2}):([0-9]{1,2}):([AP])M'
 timetable_dict_parser = re.compile(TIMETABLE_DICT_RE)
 
-OUTPUT_FILENAME = "timetable.ics"
+if not os.path.exists("data.txt"):
```
Comment (icyflame) [inline generate_ics.py]: Hey, that's a whole lot of lines do something very simple. Just keep it very very simple. I am sorry to keep requesting changes, but there's too much cruft here. :slightly_smiling_face: Here's a way to do it:

```py
INPUT_FILE = args.input if args.input else "data.txt"
if not os.path.exists(INPUT_FILE):
    print "Input file %s does not exist." % (INPUT_FILE)
    os.exit(1)
```

**Comment by rohitner** [comment]
can this be merged now?

**Comment by nishnik** [comment]
+1 from my side. Ping @icyflame

**Pull request merged**